plugins/packages/restapi/lib/index.ts

```diff
@@ -61,20 +61,31 @@ export default class RestapiQueryService implements QueryService {
     return Object.fromEntries(urlParams);
   }
 
+  customParams(sourceOptions: any) {
+    const customParams = Object.fromEntries(sourceOptions['custom_auth_params']);
+    Object.keys(customParams).forEach((key) => (customParams[key] === '' ? delete customParams[key] : {}));
+    return customParams;
+  }
+
   async run(sourceOptions: any, queryOptions: any, dataSourceId: string): Promise<RestAPIResult> {
     /* REST API queries can be adhoc or associated with a REST API datasource */
     const hasDataSource = dataSourceId !== undefined;
     const requiresOauth = sourceOptions['auth_type'] === 'oauth2';
 
     const headers = this.headers(sourceOptions, queryOptions, hasDataSource);
+    const customParams = this.customParams(sourceOptions);
 
     /* Chceck if OAuth tokens exists for the source if query requires OAuth */
     if (requiresOauth) {
       const tokenData = sourceOptions['tokenData'];
 
       if (!tokenData) {
         const tooljetHost = process.env.TOOLJET_HOST;
-        const authUrl = `${sourceOptions['auth_url']}?response_type=code&client_id=${sourceOptions['client_id']}&redirect_uri=${tooljetHost}/oauth2/authorize&scope=${sourceOptions['scopes']}`;
+        const authUrl = `${sourceOptions['auth_url']}?response_type=code&client_id=${
+          sourceOptions['client_id']
+        }&redirect_uri=${tooljetHost}/oauth2/authorize&scope=${sourceOptions['scopes']}&access_type=offline${
+          customParams['prompt'] ? '&prompt=consent' : ''
```

Comment:
I think rather than hardcoding by checking for presence, it will be better to simply append all custom auth params as query params?

Reply:
But in Ui we don't currently have a separation b/w what's sent as json body and query params. But that can be taken up to be fixed separately, and invalid params would be ignored. Might need to check that.

Reply:
So now go with append custom auth params right. we are also providing the custom params to JSON body in fetchOauthToken function. maybe it will ignore as you said.

Reply:
Yeah. Let's check it against google oauth endpoint once. We can iterate upon the UI to fix this separately if so.